pkg/spoke/controllers/manifestwork_controller.go

```diff
@@ -280,9 +280,16 @@ func (m *ManifestWorkController) applyUnstructrued(data []byte, recorder events.
 		return nil, false, err
 	}
 
-	// Do not update unstructured type right now. Need to figure out how to merge.
-
-	return existing, false, err
+	// Compare and update the unstrcuctured.
+	if isSameUnstructured(required, existing) {
+		return existing, false, nil
+	}
+	required.SetResourceVersion(existing.GetResourceVersion())
+	actual, err := m.spokeDynamicClient.Resource(gvr).Namespace(required.GetNamespace()).Update(
+		context.TODO(), required, metav1.UpdateOptions{})
+	recorder.Eventf(fmt.Sprintf(
```

Comment:
minor: if you get an error, make a different warning event that shows the error.